controllers/dscinitialization/observibility_setup.go

```diff
@@ -0,0 +1,157 @@
+// +kubebuilder:rbac:groups="monitoring.rhobs",resources=alertmanagerconfigs,verbs=get;create;patch;update;delete
+
+// +kubebuilder:rbac:groups="monitoring.rhobs",resources=monitoringstacks,verbs=get;create;patch;delete;deletec
+
+// currently the logic is only called if it is for downstream.
+func (r *DSCInitializationReconciler) configureObservability(instance *dsciv1.DSCInitialization) error {
+	switch instance.Spec.Monitoring.ManagementState {
+	case operatorv1.Managed:
+		capabilities := []*feature.HandlerWithReporter[*dsciv1.DSCInitialization]{
+			r.observabilityCapability(instance, oboCondition(status.ConfiguredReason, "CMO is configured")),
+		}
+		for _, capability := range capabilities {
+			capabilityErr := capability.Apply()
+			if capabilityErr != nil {
+				r.Log.Error(capabilityErr, "failed applying ClusterObservability resources")
+				r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed applying ClusterObservability")
+				return capabilityErr
+			}
+		}
+	case operatorv1.Removed:
+		r.Log.Info("existing ClusterObservability CR (owned by operator) will be removed")
+		if err := r.removeObservability(instance); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *DSCInitializationReconciler) removeObservability(instance *dsciv1.DSCInitialization) error {
+	if instance.Spec.Monitoring.ManagementState == operatorv1.Managed {
+		capabilities := []*feature.HandlerWithReporter[*dsciv1.DSCInitialization]{
+			r.observabilityCapability(instance, oboCondition(status.RemovedReason, "ClusterObservability resources removed")),
+		}
+
+		for _, capability := range capabilities {
+			capabilityErr := capability.Delete()
+			if capabilityErr != nil {
+				r.Log.Error(capabilityErr, "failed deleting Clusterobservability resources")
+				r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed deleting Clusterobservability resources")
+
+				return capabilityErr
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *DSCInitializationReconciler) observabilityCapability(instance *dsciv1.DSCInitialization, initialCondition *conditionsv1.Condition) *feature.HandlerWithReporter[*dsciv1.DSCInitialization] { //nolint:lll // Reason: generics are long
+	return feature.NewHandlerWithReporter(
+		feature.ClusterFeaturesHandler(instance, r.observabilityCapabilityFeatures(instance)),
+		createCapabilityReporter(r.Client, instance, initialCondition),
+	)
+}
+
+func (r *DSCInitializationReconciler) observabilityCapabilityFeatures(instance *dsciv1.DSCInitialization) feature.FeaturesProvider {
+	return func(handler *feature.FeaturesHandler) error {
+		monitoringNamespace := instance.Spec.Monitoring.Namespace
+		alertmanageErr := feature.CreateFeature("create-alertmanager").
+			For(handler).
+			ManifestSource(Templates.Source).
+			Manifests(
+				path.Join(Templates.AlertManageDir),
+			).
+			WithData(obo.AlertmanagerDataValue). // fill in alertmanager data
```

Comment:
one issue I found out here(for the config map) is that, the `WithData()` where we inject the template from the configmap is called before the manifests are applied. So I we wont have the configmap ready when withData is called. Alternative solution could be to apply the configmap in precondition?

Reply:
i saw this part yesterday but not done anything to fix it yet. 
i am still thinking

Reply:
this https://github.com/opendatahub-io/opendatahub-operator/pull/1004/commits/3c39f4fcff076e0b4697c7b5574ae21e505420c1 should have it fixed as an ugly workaround.

Reply:
cool!